webapp/src/store/index.js

```diff
@@ -423,7 +429,48 @@ export default createStore({
       return state.userActivityCache[cacheKey];
     },
   },
-  actions: {},
+  actions: {
+    async fetchCurrentUser({ state, commit }, { fullInfo = false } = {}) {
+      if (state.currentUserInfoPromise) {
+        return state.currentUserInfoPromise;
+      }
+
+      if (!fullInfo && state.currentUserInfoLoaded) {
+        return state.currentUserID !== null;
+      }
+
+      commit("setCurrentUserInfoLoading", true);
+
+      const promise = fetch(`${API_URL}/get-current-user/`, {
```

Comment:
Sorry to be a pain but I'd rather all the direct API fetches are done in `server_fetch_utils`, we likely need to use the `fetch_get` wrapper there anywhere.

Could you put this back there and manage the store state upon request?